src/modules/staking/services/staking.filtering.service.ts

```diff
@@ -43,4 +41,36 @@ export class StakingFilteringService {
 
         return filteredAddresses;
     }
+
+    async stakingFarmsByRewardsDepleted(
+        stakingFilter: StakingFarmsFilter,
+        stakeAddresses: string[],
+    ): Promise<string[]> {
+        if (
+            typeof stakingFilter.rewardsDepleted === 'undefined' ||
+            stakingFilter.rewardsDepleted === null
+        ) {
+            return stakeAddresses;
+        }
+
+        const allProduceRewardsEnabled =
```

Comment:
you are awaiting each service call one after another,  you could parallelize these service calls using Promise.all to improve performance